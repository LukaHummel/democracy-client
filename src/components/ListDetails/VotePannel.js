import React from 'react';
import styled from 'styled-components/native';
import Ionicons from 'react-native-vector-icons/Ionicons';
import { Alert } from 'react-native';

const Wrapper = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  height: 132;
  background-color: rgba(250, 250, 250, 0.9);
`;

const CircleButton = styled.TouchableOpacity.attrs({
  onPress: () => () => Alert.alert('du hast nun abgestimmt'),
})`
  align-items: center;
  justify-content: center;
  width: 87.5;
  height: 87.5;
  border-radius: 50;
  border-width: 1;
`;

const YesButton = styled(CircleButton)`
  border-color: #44db5e;
  background-color: rgb(68, 219, 94);
`;
const YesIcon = styled(Ionicons).attrs({
  name: 'ios-thumbs-up-outline',
  size: 50,
})``;

const NoButton = styled(CircleButton)`
  border-color: #fe3824;
  background-color: rgb(254, 56, 36);
`;
const NoIcon = styled(Ionicons).attrs({
  name: 'ios-thumbs-down-outline',
  size: 50,
})``;

const AbstentionButton = styled(CircleButton)`
  width: 63.5;
  height: 63.5;
  border-color: rgba(0, 118, 255, 0.71);
  background-color: rgba(0, 118, 255, 0.71);
`;
const AbstentionIcon = styled(Ionicons).attrs({
  name: 'ios-pause-outline',
  size: 30,
})`
  transform: rotate(90deg);
`;

export default () => (
  <Wrapper>
    <YesButton>
      <YesIcon />
    </YesButton>
    <AbstentionButton>
      <AbstentionIcon />
    </AbstentionButton>
    <NoButton>
      <NoIcon />
    </NoButton>
  </Wrapper>
);
