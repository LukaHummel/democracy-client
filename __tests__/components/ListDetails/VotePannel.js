import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import VotePannel from '../../../src/components/ListDetails/VotePannel';

describe('test components/List/Row', () => {
  it('render row', () => {
    const wrapper = renderer.create(<VotePannel />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
