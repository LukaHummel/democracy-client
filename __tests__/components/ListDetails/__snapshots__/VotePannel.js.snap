// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components/List/Row render row 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(250, 250, 250, 0.9)",
        "flexDirection": "row",
        "height": 132,
        "justifyContent": "space-around",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(68, 219, 94)",
        "borderColor": "#44db5e",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 87.5,
        "justifyContent": "center",
        "opacity": 1,
        "width": 87.5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 50,
          },
          Array [
            Object {},
            undefined,
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 118, 255, 0.71)",
        "borderColor": "rgba(0, 118, 255, 0.71)",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 63.5,
        "justifyContent": "center",
        "opacity": 1,
        "width": 63.5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 30,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "90deg",
                },
              ],
            },
            undefined,
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(254, 56, 36)",
        "borderColor": "#fe3824",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 87.5,
        "justifyContent": "center",
        "opacity": 1,
        "width": 87.5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 50,
          },
          Array [
            Object {},
            undefined,
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
